plutus-ledger-api/src/PlutusLedgerApi/V3/Contexts.hs

```diff
@@ -0,0 +1,577 @@
+{-# LANGUAGE DerivingVia       #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards   #-}
+{-# LANGUAGE TemplateHaskell   #-}
+{-# OPTIONS_GHC -Wno-simplifiable-class-constraints #-}
+{-# OPTIONS_GHC -fno-omit-interface-pragmas #-}
+{-# OPTIONS_GHC -fno-specialise #-}
+{-# OPTIONS_GHC -fno-strictness #-}
+
+module PlutusLedgerApi.V3.Contexts where
+
+import GHC.Generics (Generic)
+import Prettyprinter (nest, vsep, (<+>))
+import Prettyprinter.Extras
+
+import PlutusLedgerApi.V2 qualified as V2
+import PlutusTx qualified
+import PlutusTx.AssocMap hiding (filter, mapMaybe)
+import PlutusTx.Prelude qualified as PlutusTx
+
+import Prelude qualified as Haskell
+
+newtype ColdCommitteeHash = ColdCommitteeHash V2.Credential
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+  deriving (Pretty) via (PrettyShow ColdCommitteeHash)
+  deriving newtype
+    ( PlutusTx.Eq
+    , PlutusTx.ToData
+    , PlutusTx.FromData
+    , PlutusTx.UnsafeFromData
+    )
+
+newtype HotCommitteeHash = HotCommitteeHash V2.Credential
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+  deriving (Pretty) via (PrettyShow HotCommitteeHash)
+  deriving newtype
+    ( PlutusTx.Eq
+    , PlutusTx.ToData
+    , PlutusTx.FromData
+    , PlutusTx.UnsafeFromData
+    )
+
+newtype DRepHash = DRepHash V2.Credential
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+  deriving (Pretty) via (PrettyShow DRepHash)
+  deriving newtype
+    ( PlutusTx.Eq
+    , PlutusTx.ToData
+    , PlutusTx.FromData
+    , PlutusTx.UnsafeFromData
+    )
+
+data DRep
+  = DRepCredential DRepHash
+  | DRepAlwaysAbstain
+  | DRepAlwaysNoConfidence
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+  deriving (Pretty) via (PrettyShow DRep)
+
+instance PlutusTx.Eq DRep where
+  {-# INLINEABLE (==) #-}
+  DRepCredential a == DRepCredential a'            = a PlutusTx.== a'
+  DRepAlwaysAbstain == DRepAlwaysAbstain           = Haskell.True
+  DRepAlwaysNoConfidence == DRepAlwaysNoConfidence = Haskell.True
+  _ == _                                           = Haskell.False
+
+data Delegatee
+  = DelegStake V2.PubKeyHash
+  | DelegVote DRep
+  | DelegStakeVote V2.PubKeyHash DRep
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+  deriving (Pretty) via (PrettyShow Delegatee)
+
+instance PlutusTx.Eq Delegatee where
+  {-# INLINEABLE (==) #-}
+  DelegStake a == DelegStake a' = a PlutusTx.== a'
+  DelegVote a == DelegVote a' = a PlutusTx.== a'
+  DelegStakeVote a b == DelegStakeVote a' b' =
+    a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+  _ == _ = Haskell.False
+
+{- | There is probably no need for Plutus Scripts to even know about all this data in Anchors,
+but it is here for completeness. So we might wanna omit them, especially since urls are a
+bit tricky, since we can't guarantee their validity, not even that they are utf8-encoded.
+-}
+data Anchor = Anchor
+  { anchorUrl      :: PlutusTx.BuiltinString -- Arbitrary Url
```

Comment:
This is just a `ByteString` on the ledger side, so maybe using String here is not a good idea. I chose String in the ticket just to emphasize that it is textual data, but we do nothing to validate it.